### PR TITLE
fixes https://github.com/PyBites-Open-Source/pybitesbooks/issues/61

### PR DESCRIPTION
**Pull request by bbelderbos**

### CODE REVIEW

**Comment by bbelderbos** [comment]
Wrote a manage.py command (not included) to reproduce the issue - bug:

Screenshot 2023-06-06 at 16 05 17

Now with this code + same data:

Screenshot 2023-06-06 at 16 27 28

Adding a few more pomodori to test the "meet goal" scenario:

Screenshot 2023-06-06 at 16 29 33

TODO: add a pytest ...

**Comment by bbelderbos** [comment]
@greywidget correct, I wanted to git ignore more patterns. No rush, this feature is totally isolated and not critical and not often used. I did it more as an exercise, as well as working more with pytest-django. Thanks.

File: pomodoro/views.py [hunk 1]
```diff
@@ -16,32 +16,16 @@ def track_pomodoro(request):
...
-                   'pomo_minutes': DEFAULT_POMO_MIN,
-                   'pomos_done': pomos_done,
-                   'min_read': min_read,
-                   'pomos_todo': pomos_todo
-                   })
+    context = {
+        'week_stats': sorted(week_stats.items(), reverse=True),
+        'week_goal': DEFAULT_POMO_GOAL,
+    }
+    return render(request, 'pomodoro/pomodoro.html', context)
```
Comment (greywidget) [inline pomodoro/views.py:31]: I think pomo_minutes is still required

File: tests/test_pomodoro.py
Comment (greywidget) [inline tests/test_pomodoro.py:1]: tests look OK to me, all passing

**Review by greywidget — CHANGES_REQUESTED** [review]
- pomo_minutes needs to be passed from view or omitted from template
- Not certain if the intention was to remove the text against each pomo e.g
 `(7 pomodori / 2.92 hours read)`

File: .gitignore
[the remark this answers is not in the document]
Comment (bbelderbos) [reply .gitignore:1]: I even made an alias :)

```
$ alias gitign
gitign='curl -s https://raw.githubusercontent.com/github/gitignore/main/Python.gitignore > .gitignore'
```

File: myreadinglist/templatetags/tags.py [hunk 2]
```diff
@@ -39,9 +39,19 @@ def user2rgb(userid):
...
     idx = userid % len(COLORS)
     bg, fg = COLORS[idx]
-    return f'background-color: {bg}; color: {fg};'
+    return f"background-color: {bg}; color: {fg};"
 
 
 @register.filter
 def unslugify(value):
-    return value.replace('-', ' ')
+    return value.replace("-", " ")
```
[the remark this answers is not in the document]
Comment (bbelderbos) [reply myreadinglist/templatetags/tags.py:47]: Opened issue.

File: myreadinglist/test_settings.py
[the remark this answers is not in the document]
Comment (bbelderbos) [reply myreadinglist/test_settings.py:1]: Realized on code clinic that was a missed opportunity here, then Ryan shared his settings, no brainer :)

File: pomodoro/templates/pomodoro/pomodoro.html
[the remark this answers is not in the document]
Comment (bbelderbos) [reply pomodoro/templates/pomodoro/pomodoro.html:1]: I don't like that Django templates fail silently. pomo_minutes is only in the template, not passed in indeed, will fix

File: pomodoro/templates/pomodoro/pomodoro.html [hunk 3]
```diff
@@ -37,14 +38,18 @@ <h3><blockquote class="mui--text-display1">{{ week_goal }} x {{ pomo_minutes }}
...
-        {% for yyww, pomodori in week_stats %}
+        {% for week, pomodori in week_stats %}
         <tr>
-          <td>{{ yyww }} | {% if pomodori|length >= week_goal %}😄{% else %}😭{% endif %}</td>
+          <td>{{ week }} | {% if pomodori >= week_goal %}😄{% else %}😭{% endif %}</td>
           <td>
-            {% for _ in pomodori %}🍅{% endfor %}
-            {% for _ in pomos_todo %}📖{% endfor %}
-            <small>(<strong>{{ pomos_done }}</strong> pomodori / 
-              <strong>{{ min_read }}</strong> hours read)</small>
```
[the remark this answers is not in the document]
Comment (bbelderbos) [reply pomodoro/templates/pomodoro/pomodoro.html:47]: I realized this later, I can put it back, I think it's nice. What do you think? Put back or drop?

File: tests/test_pomodoro.py
Comment (bbelderbos) [reply tests/test_pomodoro.py:1]: ChatGPT helped a lot with this :)

Thanks for reviewing!

**Comment by bbelderbos** [comment]
> @bbelderbos If we do want to keep the text we could append hours to the `week_stats` in `views.py`:
> 
> * add `DEFAULT_POMO_MIN` back to the imports
> 
> ```
>     week_stats = Counter(pomo.week for pomo in pomodori)
>     # Add in hours read
>     week_stats = [
>         (week, count, round((count * DEFAULT_POMO_MIN) / 60, 2))
>         for week, count in week_stats.items()
>     ]
> 
>     context = {
>         "week_stats": sorted(week_stats, reverse=True),
>         "week_goal": DEFAULT_POMO_GOAL,
>     }
>     return render(request, "pomodoro/pomodoro.html", context)
> ```
> 
> And in the template:
> 
> ```
>     <tbody>
>         {% for week, pomodori, hours_read in week_stats %}
>         <tr>
>           <td>{{ week }} | {% if pomodori >= week_goal %}😄{% else %}😭{% endif %}</td>
>           <td>
>             {% filter multiply:pomodori %}
>               🍅
>             {% endfilter %}
>             {% with week_goal|subtract:pomodori as left %}
>               {% filter multiply:left %}
>                 📖
>               {% endfilter %}
>             {% endwith %}
>             <small>(<strong>{{ pomodori }}</strong> pomodori / 
>               <strong>{{ hours_read }}</strong> hours read)</small>
>           </td>
>         </tr>
>         {% endfor %}
>     </tbody>
> ```
> 
> But I don't feel strongly about it - the graphic alone, without text is nice and clean.
> 
> By the way, I meant to say - the pomodoro `views.py` is much cleaner now, good job

Thank you! Yeah maybe we just leave it icons only, it's clean and shows most importantly if you've hit the goal.

File: pomodoro/templates/pomodoro/pomodoro.html
[the remark this answers is not in the document]
Comment (bbelderbos) [reply pomodoro/templates/pomodoro/pomodoro.html:1]: https://github.com/PyBites-Open-Source/pybitesbooks/pull/75/commits/99455fd3712d47e71f550c6fd10ae843fe4692b0

File: pomodoro/templates/pomodoro/pomodoro.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (bbelderbos) [reply pomodoro/templates/pomodoro/pomodoro.html:47]: Keep dropped then.

File: pomodoro/migrations/0002_alter_pomodoro_end.py
[the remark this answers is not in the document]
Comment (greywidget) [reply pomodoro/migrations/0002_alter_pomodoro_end.py:1]: Ha, I should have realised this - I hit the same thing in my testing, had to create then update the Pomodori

**Pull request merged**